src/Service/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -506,5 +493,58 @@ public void ProcessOdataClause(FilterClause odataClause)
         {
             return filterClause.Expression.Accept<string>(visitor);
         }
+
+        /// <summary>
+        /// Gets the value of the parameter cast as the system type
+        /// </summary>
+        protected object GetParamAsSystemType(string param, string paramName, Type systemType)
+        {
+            try
+            {
+                return ParseParamAsSystemType(param, systemType);
+            }
+            catch (Exception e)
+            {
+                if (e is FormatException ||
+                    e is ArgumentNullException ||
+                    e is OverflowException)
+                {
+                    if (MetadataProvider.IsDevelopmentMode())
+                    {
+                        if (MetadataProvider.EntityToDatabaseObject[EntityName].SourceType is SourceType.StoredProcedure)
+                        {
+                            throw new DataApiBuilderException(
+                                message: $@"Parameter ""{param}"" cannot be resolved as stored procedure parameter ""{paramName}"" " +
+                        $@"with type ""{systemType.Name}"".",
+                                statusCode: HttpStatusCode.BadRequest,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest,
+                                innerException: e
+                                );
+                        }
+                        else
+                        {
+                            throw new DataApiBuilderException(
+                                message: $"Parameter \"{param}\" cannot be resolved as column \"{paramName}\" " +
+                                    $"with type \"{systemType.Name}\".",
+                                statusCode: HttpStatusCode.BadRequest,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest,
+                                innerException: e);
+                        }
+
+                    }
+                    else
+                    {
+                        throw new DataApiBuilderException(
+                                message: $"Invalid value provided for field: {paramName}",
+                                statusCode: HttpStatusCode.BadRequest,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest,
+                                innerException: e);
+                    }
```

Comment:
Can actually throw exception here instead of throwing it at 3 different places. Only the exception message is different among the 3 instances. So, can just evaluate the message using the if/else block.

Reply:
Updated to determine the error message conditionally and moved the actual throwing of exception to the end.